src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,62 @@
+package duke;
+
+import duke.task.Task;
+
+import java.util.Scanner;
+
+/**
+ * Encapsulates a Ui object to handle user interaction.
+ */
+public class Ui {
+
+    private Scanner sc;
+
+    public Ui() {
+        sc = new Scanner(System.in);
+    }
+
+    public void showWelcome() {
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("Hello from\n" + logo);
+        showLine();
+        System.out.println("     Hello! I'm Duke");
+        System.out.println("     What can I do for you?");
+        showLine();
+    }
+
+    public void showLoadingError() {
+        System.out.println("File cannot be loaded. A new task list is created.");
+    }
+
+    public void showError(String msg) {
+        showLine();
+        System.out.println("     " + msg);
+        //showLine();
+    }
```

Comment:
I felt that it was good that you specify the names of the errors when you throw the DukeException so that they can be printed out using the showError method. It really helps users find out what is the specific cause of the the exceptions thrown.